src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -45,6 +45,25 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
       }
     }
 
+    //
+    // Cleaners
+    //
+
+    const asEdgeToken = (tokenId: string) => {
+      const edgeToken = wallet.currencyConfig.allTokens[tokenId]
+      if (edgeToken == null) throw new Error(`Unable to find token on wallet for ${tokenId} tokenId`)
+      return edgeToken
+    }
+    const asTokenAddress = (token: EdgeToken) => {
+      const tokenAddress = asMaybe(asTokenContractAddress)(token)
+      if (tokenAddress == null) throw new Error(`Unable to find contract address for ${token.displayName} (${token.currencyCode})`)
+      return tokenAddress
+    }
+    const asTokenIdParam = (tokenId?: string) => {
+      if (tokenId == null) throw new Error('Getting wrapped native token not supported yet. ' + 'Explicitly pass in tokenId for the wrapped token.')
+      return tokenId
+    }
```

Comment:
These are not cleaners, since they take typed input. These are utility methods, and pretty obfuscated ones at that. Please, do it the way I asked in the previous review:

```js
getContractAddress(wallet: EdgeCurrencyWallet, tokenId?: string): string | void {
  if (tokenId == null) return
  const token = wallet.currencyConfig.allTokens[tokenId]
  if (token == null) return
  const address = asMaybe(asTokenContractAddress)(token)
  return address
}
```

Now that we have a clear utility, using it is easy we can do:

```js
async getAprQuote(tokenId?: string): Promise<number> {
  const address = getContractAddress(wallet, tokenId)
  if (address == null) {
    throw new Error(`Unable to find contract address for token ${tokenId} on AAVE`)
  }

  // ...
}
```

Nobody is supposed to pass bad data to AAVE, so the `throw` won't happen unless we have a bug. Therefore, we don't need to have a bunch of over-engineered error handling logic. Just keep it simple, and return something a dev can understand.

Ok, if you *really* like over-detailed errors, I suppose you could have `getContractAddress` throw when `token == null` or when the cleaner fails, but this wouldn't actually include any additional information, so I don't see the point...

Reply:
I agree. I like this better. Originally, I had these utilities named as "resolveEdgeToken" and "resolveTokenAddress" etc. Then incorrectly conflated that with cleaners.